src/gnc_attitude_estimator.cpp

```diff
@@ -282,15 +282,15 @@ static void ukf(AttitudeEstimatorState &state, AttitudeEstimatorData const &data
     // Plus the associated covariances
     UkfVector6 dx1 = state.sigmas[0] - x_bar;
     UkfVector5 dz1 = state.measures[0] - z_bar;
-    P_bar = P_bar + weight_c * dx1 * lin::transpose(dx1);
-    P_vv = P_vv + weight_c * dz1 * lin::transpose(dz1);
-    P_xy = P_xy + weight_c * dx1 * lin::transpose(dz1);
+    P_bar = (weight_c * dx1) * lin::transpose(dx1);
+    P_vv = (weight_c * dz1) * lin::transpose(dz1);
+    P_xy = (weight_c * dx1) * lin::transpose(dz1);
     for (lin::size_t i = 1; i < 13; i++) {
       dx1 = state.sigmas[i] - x_bar;
       dz1 = state.measures[i] - z_bar;
-      P_bar = P_bar + weight_o * (dx1 * lin::transpose(dx1));
-      P_vv = P_vv + weight_o * (dz1 * lin::transpose(dz1));
-      P_xy = P_xy + weight_o * (dx1 * lin::transpose(dz1));
+      P_bar = P_bar + (weight_o * dx1) * lin::transpose(dx1);
+      P_vv = P_vv + (weight_o * dz1) * lin::transpose(dz1);
+      P_xy = P_xy + (weight_o * dx1) * lin::transpose(dz1);
```

Comment:
Also, explicitly multiplying the scalar on the vector as opposed to on the matrix to reduce floating point multiplications.